### ISSUE TITLE
ff_omardeen wrong druid credit count (on esdraelon.amtgard.com)

### ISSUE THREAD

**Issue by amtgard** [amtgard/ORK3#92]
ff_omardeen wrong druid credit count (on esdraelon.amtgard.com)

Shows 4 Druid rather than 5 Druid credits.  This may be by design.

**Comment by Glenalth** [comment]
Unsure if related. Seeing a total of 2 Anti-Pal credits on my own record (Glenalth), with only one credit on attendance record that I can see.

**Comment by amtgard** [comment]
Shit.  That's even worse.

**Comment by amtgard** [comment]
OK, I purposely or inadvertently mis-created the unique key on this table.  There should be a unique key on mundane_id/date.  My guess is that there was in fact a key, and that I had to remove it in order to accommodate the ORK 2 import. 

Either way, both ff_omardeen and your account suffer from the same issue, namely, you have attendance entered at different locations, or the same location with different classes, on the same day.  Depending on the specifics, primarily the ordering the records on disk and the gods of MySQL, this resulted in different grouping mechanisms.

For Glenalth's account, this results in a Paladin credit being grouped into an AP credit for one day, and for ff_omardeen, it groups a Druid credit into a Warrior credit.

The correct fix is to consider only one entry per day, and to throw away the extra entry in some manner.  It's too late to apply the key in a reasonable way, so I have opted to have the SQL simply keep only the "first" entry for any given day (using the autonumber primary key on the attendance table as the arbiter for "first").
